shoplist/src/main/java/com/netguru/codereview/network/ShopListApiMock.kt

```diff
@@ -0,0 +1,42 @@
+package com.netguru.codereview.network
+
+import com.netguru.codereview.network.model.ShopListItemResponse
+import com.netguru.codereview.network.model.ShopListResponse
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class ShopListApiMock : ShopListApi {
+    override suspend fun getShopLists(): List<ShopListResponse> =
+        coroutineScope {
+            List(9999) { index ->
```

Comment:
Although this is a mock API, as a generic rule of thumb, I would suggest paginating the request instead of asking for 9999 elements to the API. Basically asking for API support for parameters `page: Int` and `size: Int` and ask for page:0, size:20 and after that lazy load the next items by requests page:1, page:2 and so on